Orchestration/OrchestrationService/ModuleConfigurationDeployment.ps1

```diff
@@ -474,6 +413,326 @@ Function New-Deployment {
+            
+            # Finished first loop, let's wait for a minute
+            # to give time for the resource group to get deleted
+            if ($allResourceGroupsDeleted -eq $false) {
+                Write-Debug "Starting sleep"
+                Start-Sleep -Seconds 60
+            }
+        }
+    }
+    catch {
+        Write-Host "An error ocurred while running TearDownEnvironment";
+        Write-Host $_;
+        throw $_;
+    }
+}
+
+Function Start-Init {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$false)]
+        [string]
+        $ArchetypeInstanceName,
+        [Parameter(Mandatory=$true)]
+        [string]
+        $DefinitionPath,
+        [Parameter(Mandatory=$false)]
+        [string]
+        $WorkingDirectory
+    )
+    try {
+        $defaultWorkingDirectory = `
+            Get-WorkingDirectory `
+                -WorkingDirectory $WorkingDirectory;
+
+        Write-Debug "Working directory is: $defaultWorkingDirectory";
+
+        $global:factory = `
+            Invoke-Bootstrap `
+                -WorkingDirectory $defaultWorkingDirectory;
+
+        $global:deploymentService = `
+            $factory.GetInstance('IDeploymentService');
+
+        $global:cacheDataService = `
+            $factory.GetInstance('ICacheDataService');
+
+        $global:auditDataService = `
+            $factory.GetInstance('IDeploymentAuditDataService');
+
+        $global:moduleStateDataService = `
+            $factory.GetInstance('IModuleStateDataService');
+
+        $global:customScriptExecution = `
+            $factory.GetInstance('CustomScriptExecution');
+
+        # Contruct the archetype instance object only if it is not already
+        # cached
+        $archetypeInstanceJson = `
+            New-ConfigurationInstance `
+                -FilePath $DefinitionPath `
+                -WorkingDirectory $defaultWorkingDirectory `
+                -CacheKey $ArchetypeInstanceName;
+        
+        $location = ''
+
+        # Check for invariant
+        if ($null -eq $archetypeInstanceJson.Parameters.Location) {
+            throw "Location value is not present in the archetype parameters file"
+        }
+        else {
+            $location = $archetypeInstanceJson.Parameters.Location
+        }
+
+        # Retrieve the Archetype instance name if not already passed
+        # to this function
+        $archetypeInstanceName = `
+            Get-ArchetypeInstanceName `
+                -ArchetypeInstance $archetypeInstanceJson `
+                -ArchetypeInstanceName $ArchetypeInstanceName;
+
+        return @{
+            WorkingDirectory = $defaultWorkingDirectory
+            ArchetypeInstanceJson = $archetypeInstanceJson
+            ArchetypeInstanceName = $archetypeInstanceName
+            Location = $location
+        }
+    }
+    catch {
+        Write-Host "An error ocurred while running Start-Init";
+        Write-Host $_;
+        throw $_;
+    }
+}
+
+Function Get-AllModules {
```

Comment:
I am trying to encapsulate the implementation and just have a generic name, the reason is because if we get to implement the parallel job execution, most likely the data structure that we return will be different than just a simple array. Thoughts?